lib/ansible/plugins/callback/foreman.py

```diff
@@ -134,26 +134,37 @@ def _ssl_verify(self):
             verify = self.FOREMAN_SSL_VERIFY
         return verify
 
-    def send_facts(self, host, data):
+    def send_facts(self):
         """
         Sends facts to Foreman, to be parsed by foreman_ansible fact
         parser.  The default fact importer should import these facts
         properly.
         """
-        data["_type"] = "ansible"
-        data["_timestamp"] = datetime.now().strftime(self.TIME_FORMAT)
-        facts = {"name": host,
-                 "facts": data,
-                 }
-        try:
-            r = requests.post(url=self.FOREMAN_URL + '/api/v2/hosts/facts',
-                              data=json.dumps(facts),
-                              headers=self.FOREMAN_HEADERS,
-                              cert=self.FOREMAN_SSL_CERT,
-                              verify=self.ssl_verify)
-            r.raise_for_status()
-        except requests.exceptions.RequestException as err:
-            print(to_text(err))
+        for host, results in self.items.items():
+            ansible_facts = {}
+            for dummy, result in results:
+                if 'ansible_facts' in result:
+                    ansible_facts.update(result['ansible_facts'])
+
+            facts = {
+                "name": host,
+                "facts": {
+                    "ansible_facts": ansible_facts,
+                    "_type": "ansible",
+                    "_timestamp": datetime.now().strftime(self.TIME_FORMAT),
```

Comment:
Are you sure that you're using the right timezone?

Reply:
I see that it's copy-pasted so you could probably fix this separately from this PR.